docs/modules/writing_a_module.md

```diff
@@ -82,4 +82,53 @@ the callback name as the argument name and the function as its value. A
 `register_[...]_callbacks` method exists for each category.
 
 Callbacks for each category can be found on their respective page of the
-[Synapse documentation website](https://matrix-org.github.io/synapse).
\ No newline at end of file
+[Synapse documentation website](https://matrix-org.github.io/synapse).
+
+## Caching
+
+Modules can leverage Synapse's caching tools to manage their own cached functions. This
+canbe helpful for modules that need to repeatedly request the same data from the database
+or a remote service.
+
+Functions that need to be wrapped with a cache need to be decorated with a `@cached()`
```

Comment:
`@cached` has sharp edges. I would like to see a warning here or at least some kind of pointer to the docstring of `@cached`... which doesn't exist.

Are there limitations on functions that can be cached? IIRC they must not have any variadic args (no `*args, **kwargs`). When should a module author use `tree=True` or `iterable=True`?

Reply:
I'm not sure I'm best positioned to be documenting `cached` given I haven't used it with arguments in a while. Presumably in most cases just using it without arguments should be fine. Do we want to block this on `cached` getting a docstring?

Reply:
> Do we want to block this on `cached` getting a docstring?

Not especially. Back in the queue to solicit other opinions

Reply:
I'm almost inclined to really strip down the public API we expose for `@cached`. I understand why we exposed it to modules, but I'm *very* cautious about making all the various options a public API.

Reply:
Looks like this thread was done, so I'm going to resolve it.